### PR TITLE
Enforce formatting

### LINKED ISSUE #86: Auto-format source code to ensure Google style compliance
Use https://github.com/google/google-java-format (specifically https://github.com/google/google-java-format/blob/master/scripts/google-java-format-diff.py) in our gradle scripts.

### PR DESCRIPTION
**Pull request by pongad**
Task test now depends on verifyGoogleJavaFormat which will fail
if any file is not properly formatted.

Run `./gradlew goJF` to format files.

Fixes #86.

### CODE REVIEW

**Comment by bjwatson** [comment]
Sorry @pongad. I missed this with the summit last week. I'll review on Monday. Can you resolve merge conflicts?

**Comment by garrettjonesgoogle** [comment]
I think this should wait on my PR https://github.com/googleapis/toolkit/pull/172 .

**Comment by bjwatson** [comment]
Sure. No problem.

**Comment by pongad** [comment]
Rebased on @garrettjonesgoogle 's merge.
@bjwatson PTAL.

**Comment by garrettjonesgoogle** [comment]
LGTM but sounds like @bjwatson  wants to take a look too.

File: build.gradle [hunk 1]
```diff
@@ -13,16 +13,24 @@ buildscript {
     }
     mavenCentral()
   }
+  repositories {
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
```
Comment (bjwatson) [inline build.gradle]: Is this canonical for Gradle to repeat the `repositories` block twice? It seems strange, but I'm not a Gradle expert.

Comment (pongad) [inline build.gradle]: We actually declare the block 3 times :smile: under different circumstances. This one is scoped under buildscript, the others are not.

Comment (bjwatson) [inline build.gradle]: Look right above it (still in the `buildscript` scope). ;)

Comment (pongad) [inline build.gradle]: I was looking at a wrong branch. Sorry about that. Will update.

**Comment by bjwatson** [comment]
I've updated google-java-format to 1.0, and applied our Eclipse plugin formatting.

PTAL @pongad and @garrettjonesgoogle.

Otherwise, LGTM.

**Comment by garrettjonesgoogle** [comment]
Travis failed...

**Comment by pongad** [comment]
Due to recent commits, I'll rebase

**Comment by bjwatson** [comment]
@pongad The rebase lost the changes I made, and the flakiness is still there. Run `g build` twice on your machine.

**Pull request merged**